### PR TITLE
Dev weixuan backend

### PR DESCRIPTION
**Pull request by weixuanseow**
see new_app.py and skills_based_role_portal.sql

### CODE REVIEW

File: backend/new_app.py [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+        return jsonify(
+            {
+                "code": 400,
+                "message": "Role with this name already exists!"
+            }
+        ), 400
+
+    # If the role doesn't exist, insert it into the database
+    new_role = Role(
+        Role_Name=role_name,
```
Comment (L1uM4rt1n) [inline backend/new_app.py:213]: will this work without the Role_ID?

File: backend/new_app.py [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+        return jsonify(
+            {
+                "code": 400,
+                "message": "Staff member with this email already exists!"
+            }
+        ), 400
+
+    # If staff member doesn't exist, insert into the database
+    new_staff = Staff(
+        Staff_FName=staff_fname,
```
Comment (L1uM4rt1n) [inline backend/new_app.py:317]: same for this, will this work without the Satff_ID?

File: backend/new_app.py [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+        App_Deadline=app_deadline,
+        Role_Department=role_department,
+        Role_Description=role_description,
+        Role_Requirements=role_requirements
+    )
+
+    try:
+        db.session.add(new_role)
+        db.session.commit()
+    except Exception as e:
```
Comment (L1uM4rt1n) [inline backend/new_app.py:224]: the Exception here is too broad, it looks like u are trying to capture the IntegrityError here, might be good to specify exactly which exception you are catching

File: backend/new_app.py [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+        Dept=dept,
+        Country=country,
+        Email=email,
+        Access_Rights=access_rights
+    )
+    
+    try:
+        db.session.add(new_staff)
+        db.session.commit()
+    except Exception as e:
```
Comment (L1uM4rt1n) [inline backend/new_app.py:328]: same here, would be good to specify the IntegrityError instead of a general exception

**Review by L1uM4rt1n — COMMENTED** [review]
largely looks fine, i havent tested out the endpoints on postman myself. just some comments:
1. remove the archive folder from your next commit so that it wont be merged to main
2. check for the missing ID and the raising of more specific exceptions

File: backend/new_app.py (hunk 1 above)
Comment (weixuanseow) [reply backend/new_app.py:213]: oyes need the role_ID

File: backend/new_app.py (hunk 2 above)
Comment (weixuanseow) [reply backend/new_app.py:317]: this staff_ID is auto-increment so the sql will auto add the id when we add new staff

File: backend/new_app.py (hunk 1 above)
Comment (weixuanseow) [reply backend/new_app.py:213]: i didnt autoincrement this bc i wanted to differentiate between the staff_ID so staff_ID is 1,2,3,4,... but Role_ID is 1000001, 1000002, ...

**Pull request closed without merging**